go/teams/service_helper.go

```diff
@@ -987,6 +987,58 @@ func MemberRoleFromID(ctx context.Context, g *libkb.GlobalContext, teamID keybas
 	return role, err
 }
 
+func RemoveMembers(ctx context.Context, g *libkb.GlobalContext, teamID keybase1.TeamID, users []keybase1.RemoveMemberArgs) (res keybase1.TeamRemoveMembersResult, err error) {
+	teamGetter := func() (*Team, error) {
+		return GetForTeamManagementByTeamID(ctx, g, teamID, false)
+	}
+
+	var failedToRemove []keybase1.RemoveMemberArgs
+
+	for _, user := range users {
+		var exclusiveActions []string
+		if len(user.Username) > 0 {
+			exclusiveActions = append(exclusiveActions, "username")
+		}
+		if len(user.Email) > 0 {
+			exclusiveActions = append(exclusiveActions, "email")
+		}
+		if len(user.InviteID) > 0 {
+			exclusiveActions = append(exclusiveActions, "inviteID")
+		}
+		if len(exclusiveActions) > 1 {
```

Comment:
If more than one of them is set, then the length of the concatenation of the actions will be greater than the length of any one of them. Then you don't need a slice.
```go
keysLen := len(user.Username + user.Email + string(user.InviteID))
if keysLen > len(user.Username) && keysLen > len(user.Email) && keysLen > len(user.InviteID) {
```

Reply:
Though this way may lose the nice formatting of the slice for the log (with spaces or brackets). I just copied the logic from what was in the remove-single-member code.

Reply:
The best way i thought with the suggested approach was something like:
```
exclusiveActions := user.Username + user.Email + string(user.InviteID)
		lenExclActions := len(exclusiveActions)
		
		if lenExclActions > len(user.Username) && lenExclActions > len(user.Email) && lenExclActions > len(user.InviteID) {
			g.Log.CDebugf(ctx, "RemoveMembers: can only do 1 of [%s, %s, %s] at a time", user.Username, user.Email, user.InviteID)
			failedToRemove = append(failedToRemove, user)
			continue
		}
```

Reply:
Sure, same thing.